src/win/tty.c

```diff
@@ -1608,6 +1612,30 @@ static int uv_tty_set_cursor_visibility(uv_tty_t* handle,
   return 0;
 }
 
+static int uv_tty_set_cursor_shape(uv_tty_t* handle,
+                                        DWORD* error) {
+  CONSOLE_CURSOR_INFO cursor_info;
+
+  if (!GetConsoleCursorInfo(handle->handle, &cursor_info)) {
+    *error = GetLastError();
+    return -1;
+  }
+
+  if (handle->tty.wr.ansi_csi_argv[0] <= 2) {
```

Comment:
Check that ansi_csi_argc >= 1 to deal with cases where the argument is (erroneously?) omitted.
Or even better, add an extra argument to uv_tty_set_cursor_shape and pass `d` value from the call site.